tests/tools/test_parmetis_wrapper.py

```diff
@@ -0,0 +1,271 @@
+                with open(chunk_f_name, 'r') as f:
+                    header = f.readline()
+                    num1, num2 = header.rstrip().split(' ')
+                    assert isinstance(int(num1), int)
+                    assert isinstance(int(num2), int)
+
+        # check node_data
+        output_node_data_dir = os.path.join(output_dir, 'node_data', 'paper')
+        for feat in ['feat', 'label', 'year']:
+            for i in range(num_chunks):
+                chunk_f_name = '{}-{}.npy'.format(feat, i)
+                chunk_f_name = os.path.join(output_node_data_dir, chunk_f_name)
+                assert os.path.isfile(chunk_f_name)
+                feat_array = np.load(chunk_f_name)
+                assert feat_array.shape[0] == num_papers // num_chunks
+
+        # check edge_data
+        num_edges = {
+            'paper:cites:paper': num_cite_edges,
+            'author:writes:paper': num_write_edges,
+            'paper:rev_writes:author': num_write_edges
+        }
+        output_edge_data_dir = os.path.join(output_dir, 'edge_data')
+        for etype, feat in [
+            ['paper:cites:paper', 'count'],
+            ['author:writes:paper', 'year'],
+            ['paper:rev_writes:author', 'year']
+        ]:
+            output_edge_sub_dir = os.path.join(output_edge_data_dir, etype)
+            for i in range(num_chunks):
+                chunk_f_name = '{}-{}.npy'.format(feat, i)
+                chunk_f_name = os.path.join(output_edge_sub_dir, chunk_f_name)
+                assert os.path.isfile(chunk_f_name)
+                feat_array = np.load(chunk_f_name)
+                assert feat_array.shape[0] == num_edges[etype] // num_chunks
+
+        # Trigger ParMETIS wrapper here. 
+        schema_path = os.path.join(root_dir, 'chunked-data/metadata.json')
+        results_dir = os.path.join(root_dir, 'parmetis-data')
+        os.system('mpirun -np 2 python3 tools/distpartitioning/parmetis_wrapper.py '\
+                  '--schema {} --output {}'.format(schema_path, results_dir))
+
+        # Now add all the tests and check whether the test has passed or failed
+        # Test - 1
+        # Read parmetis_nfiles and ensure all files are present
+        parmetis_data_dir = os.path.join(root_dir, 'parmetis-data')
+        assert os.path.isdir(parmetis_data_dir)
+        parmetis_nodes_file = os.path.join(parmetis_data_dir, 'parmetis_nfiles.txt')
+        assert os.path.isfile(parmetis_nodes_file)
+
+        with open(parmetis_nodes_file, 'r') as f:
+            lines = f.readlines()
+            total_node_count = 0
+            for line in lines:
+                tokens = line.split(" ")
+                print(tokens)
+                assert len(tokens) == 3
+                print(tokens[0])
+                assert os.path.isfile(tokens[0])
+                assert int(tokens[1]) == total_node_count
+
+                #check contents of each of the nodes files here
+                with open(tokens[0], 'r') as nf:
```

Comment:
Yes. it is possible to validate each line in this file. This will introduce significant processing and will take longer for the CI to complete. This is the reason why I omitted this logic here. BTW, format you indicated is correct.

Reply:
I will leave it upto you to decide. But I prefer not checking each and everyline for correctness.

Reply:
nevermind, I added what you requested. if this creates problems we can remove this checking code to save some time.